specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+        memo: Option<String>,
+    );
+    /// Batch transfer token/s and call a method on a receiver contract. A successful
+    /// workflow will end in a success execution outcome to the callback on the MultiToken
+    /// contract at the method `mt_resolve_batch_transfer`.
+    ///
+    /// You can think of this as being similar to attaching  tokens to a
+    /// function call. It allows you to attach any Fungible or Non Fungible Token in a call to a
+    /// receiver contract.
+    ///
+    /// Requirements:
+    /// * Caller of the method must attach a deposit of 1 yoctoⓃ for security
+    ///   purposes
+    /// * Contract MUST panic if called by someone other than token owner or,
+    ///   if using Approval Management, one of the approved accounts
+    /// * The receiving contract must implement `mt_on_transfer` according to the
+    ///   standard. If it does not, MultiToken contract's `mt_resolve_batch_transfer` MUST deal
+    ///   with the resulting failed cross-contract call and roll back the transfer.
+    /// * Contract MUST implement the behavior described in `mt_resolve_batch_transfer`
+    /// * `approval_id` is for use with Approval Management extension, see
+    ///   that document for full explanation.
+    /// * If using Approval Management, contract MUST nullify approved accounts on
+    ///   successful transfer.
+    ///
+    /// Arguments:
+    /// * `receiver_id`: the valid NEAR account receiving the token.
+    /// * `token_ids`: the tokens to transfer
+    /// * `amounts`: the amount of tokens to transfer for corresponding token_id
+    /// * `approval_ids`: expected approval IDs. A number smaller than
+    ///    2^53, and therefore representable as JSON. See Approval Management
+    ///    standard for full explanation. Must have same length as token_ids
+    /// * `memo` (optional): for use cases that may benefit from indexing or
+    ///    providing information for a transfer.
+    /// * `msg`: specifies information needed by the receiving contract in
+    ///    order to properly handle the transfer. Can indicate both a function to
+    ///    call and the parameters to pass to that function.
+
+    fn mt_batch_transfer_call(
+        &mut self,
+        receiver_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+        memo: Option<String>,
+        msg: String,
+    ) -> PromiseOrValue<Vec<U128>>;
+
+    /// Get the balance of an an account given token_id. For fungible token returns back amount, for
+    /// non fungible token it returns back constant 1.
+    fn mt_balance_of(&self, owner_id: AccountId, token_id: TokenId) -> U128;
+
+    /// Get the balances of an an account given token_ids. For fungible token returns back amount, for
+    /// non fungible token it returns back constant 1. returns vector of balances corresponding to token_ids
+    /// in a 1-1 mapping
+    fn mt_balance_of_batch(&self, owner_id: AccountId, token_ids: Vec<TokenId>) -> Vec<U128>;
```

Comment:
Is there no desire to be able to get the balance of multiple different owners with the same batch balance call?

If there is a use-case for this, but the initial spec here doesn't incorporate it, should we name these methods differently in some a way so that there is 'room' in the API to add a batch balance of method later that supports multiple owners?

Reply:
There wasn't a desire at the time to get all the users of a token_id. The thought here was on trying to get familiarity and reducing the scope of implementing the spec. There could be room for multiple owners across multiple token ids, where the resulting return type is more complex. 

This would deviate however from same the more widely adopted standard in erc-1155. In this case we could put something like this in the enumeration extension, which might make sense to slice and dice tokens in a way outside of contract implementers use case. 

I think the smaller the spec is the better . That said let's bring it up to the group and see what they think if we want to make room this addition :D .